src/main/java/com/iota/iri/service/tipselection/impl/WalkValidatorImpl.java

```diff
@@ -26,6 +28,12 @@
  */
 public class WalkValidatorImpl implements WalkValidator {
 
+    public static final int MAX_CACHE_SIZE = 2_000_000;
+    //As long as tip selection is synchronized we are fine with the collection not being thread safe
+    private static final BoundedSet<Hash> FAILED_BELOW_MAX_DEPTH_CACHE = new BoundedSetWrapper<>(
```

Comment:
You're allocating a `FAILED_BELOW_MAX_DEPTH_CACHE` instance both in the impl and the interface with different evil magic numbers...

Reply:
Something got messed up with the commits I will fix it.